plugins/by-name/codeium-nvim/default.nix

```diff
@@ -15,108 +13,42 @@ lib.nixvim.plugins.mkNeovimPlugin {
     khaneliman
   ];
 
-  # TODO: added 2024-09-03 remove after 24.11
-  deprecateExtraOptions = true;
-  optionsRenamedToSettings = [
-    "configPath"
-    "binPath"
-    [
-      "api"
-      "host"
-    ]
-    [
-      "api"
-      "port"
-    ]
-    [
-      "tools"
-      "uname"
-    ]
-    [
-      "tools"
-      "uuidgen"
-    ]
-    [
-      "tools"
-      "curl"
-    ]
-    [
-      "tools"
-      "gzip"
-    ]
-    [
-      "tools"
-      "languageServer"
-    ]
-    "wrapper"
-  ];
```

Comment:
What's the motivation for removing these? The alternative of moving them to a deprecation file seems fine too, right?

Reply:
> To reduce the clutter.

I believe moving the options to `./deprecation.nix` achieves this.

> I think that it is desirable for us to maintain our set of deprecations. I am in favor of cleaning the code base from (objectively?) useless warnings.

I don't necessarily disagree. I came around to the idea during our [discussion on matrix](https://matrix.to/#/!iBrKeqhFRZaBBGodLH:matrix.org/$AhfJL9W4d4HVJc7yuv16lWTd6Zo_mi05hH5T19KQKDU?via=matrix.org&via=vhack.eu&via=familleboyer.net).

But let's move this discussion to #3181.

Shall we postpone removal for now, so it doesn't block this PR?